src/object.ts

```diff
@@ -374,7 +399,7 @@ export default class ObjectSchema<
     additions: U,
     excludes: readonly [string, string][] = [],
   ) {
-    type UIn = TypeFromShape<U, TContext>;
+    type UIn = MakePartial<TypeFromShape<U, TContext>>;
```

Comment:
what's happening here?

Reply:
Because `MakeKeysOptional` is no longer forced upon all `ObjectSchema` via the `extends`, this means `shape` needs to perform this same operation instead so it retains the same behaviors as before.

Essentially I lifted `MakeKeysOptional` up to the factories and the `shape` method.